railgun/test/integration/suite_test.go

```diff
@@ -73,9 +80,63 @@ func deployControllers(ctx context.Context, ready chan ktfkind.ProxyReadinessEve
 		u := event.URL
 		proxyReady <- u
 
-		cmd := exec.CommandContext(ctx, "go", "run", "../../main.go", "--kong-url", fmt.Sprintf("http://%s:8001", u.Hostname()))
+		// create a tempfile to hold the cluster kubeconfig that will be used for the controller
+		kubeconfig, err := ioutil.TempFile(os.TempDir(), "kubeconfig-")
+		if err != nil {
+			panic(err)
+		}
+		defer kubeconfig.Close()
+
+		// dump the kubeconfig from kind into the tempfile
+		generateKubeconfig := exec.CommandContext(ctx, "kind", "get", "kubeconfig", "--name", cluster.Name())
+		generateKubeconfig.Stdout = kubeconfig
+		generateKubeconfig.Stderr = os.Stderr
+		if err := generateKubeconfig.Run(); err != nil {
+			panic(err)
+		}
```

Comment:
Currently `kubeconfig.Close()` happens in `defer` and there's no `delete kubeconfig file`.
The result appears to be that:
- KIC reads from a file that hasn't been closed after writing (and is potentially unflushed)
- the file is never deleted

If we however `kubeconfig.Close()`d after `generateKubeconfig.Run()` and `defer`red `delete kubeconfig file`, both of the problems above would go away. Or not? :thinking:

Reply:
Contextually I wasn't bothering to delete a tempfile in favor of letting the OS deal with that (in our case, Linux containers in CI), but I can add something to remove the file anyhow.

Reply:
9f71e6e159c313cf76cd759b3313925fff89ad31 resolves this :+1: